### ISSUE TITLE
Undelegation tx still shows Claim button after click

### ISSUE THREAD

**Issue by fedekunze** [cosmos/voyager#1410]
Description:

I completed my undelegation to claim back the tokens and the undelegation tx still shows the `Claim` button.

<img width="1071" alt="screen shot 2018-10-04 at 1 02 04 pm" src="https://user-images.githubusercontent.com/31522760/46470295-6c609d80-c7d6-11e8-88ea-56e64f0f3b49.png">

Full error:

```
deliverTx failed: (262246) Msg 0 failed: 
=== ABCI Log ===
Codespace: 4
Code:      102
ABCICode:  262246
Error:    
   --= Error =--
      Data: common.FmtError{format:"no unbonding delegation found", args:[]interface {}(nil)}
      Msg Traces:
   --= /Error =-
-=== /ABCI Log ===
```

**Comment by NodeGuy** [comment]
I'm confused.  Isn't the screenshot showing the error `Ending undelegation failed`, which means the undelegation didn't complete?

**Comment by fedekunze** [comment]
It completed (hence the Ended Unbonding tx) but it’s still showing the claim button, which when you click it again throws the error

**Comment by faboweb** [comment]
Scope:
- [ ] if clicking on the claim button, and the claim is successful, refreshing the UI should not show the claim button

**Comment by jbibla** [comment]
> refreshing the UI should not show the claim button

can we start moving towards _not_ having to press the refresh button?

**Comment by faboweb** [comment]
> refreshing the UI

this means in development -> when Voyager is restarted

> can we start moving towards not having to press the refresh button?

this should be mostly the case already. I would leave a refresh button in there anyway to give users a sense of security.

**Comment by jbibla** [comment]
> I would leave a refresh button in there anyway to give users a sense of security.

nooooooooo~~! we should remove them and have proper loading images on every screen and in every component.

![giphy](https://user-images.githubusercontent.com/6021933/46756990-d37ac800-cc96-11e8-9657-28bb8b2b2432.gif)

**Comment by faboweb** [comment]
yes, plus a refresh option. which product do you know, that doesn't have a refresh option? I assume users would feel unsecure if the values they are seeing are the latest.

**Comment by faboweb** [comment]
What? Can you please go to inbox or Facebook and do a pull down to refresh?

David Braun <notifications@github.com> schrieb am Do., 11. Okt. 2018, 20:24:

**Comment by faboweb** [comment]
Well, in the apps.. My bad. But in the browser as you say, we have the
ability to refresh.

Fabian Weber <frznhope@gmail.com> schrieb am Do., 11. Okt. 2018 um
20:48 Uhr:

**Comment by jbibla** [comment]
ok you are kind of right. sometimes, you can "refresh" but this is more and more like requesting more data, not data that you should have been pushed already.

**Comment by NodeGuy** [comment]
Having to request more data in those situations is still a design wart.

**Comment by NodeGuy** [comment]
Blocked because @faboweb needs to publish the new version of tendermint/ui.

**Comment by cwgoes** [comment]
@NodeGuy Did you want anything from me here or was that an accidental assignment?

**Comment by NodeGuy** [comment]
@cwgoes It was an accident, sorry.

**Comment by faboweb** [comment]
published 0.3.1

**Comment by faboweb** [comment]
The unbonding delegation received from `/stake/delegator/{addr}` doesn't have a height associated. So we can't map transactions with active unbondingDelegations. Will check if this still is the case in the latest develop.
